### PR TITLE
code changes for bug fixes identified during mag_lsc dataset

### PR DESCRIPTION
**Pull request by kylasa**
These are the bug fixes identified during testing with 'mag_lsc' dataset. Since this is a larger dataset compared to "mag" we were running into peculiar issues as following:

When creating dgl partition objects, in the convert_partition.py, for some partitions unique node-ids in the edge list were not completely representing all the nodes in that partition. To resolve this we need to change the dgl function call appropriately to pass "num_nodes" argument in the argument list.

When creating a Tensor, for a very large integer list, torch.Tensor() function call used earlier. This was working for small integer list, but when it exceeds certain size we were noticing data corruption in the tensor generate from this function call. This was a very tricky bug to solve. Changed this function call with torch.from_numpy() to create a tensor from numpy array. This creates the target tensor without any data corruption.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented
- [X] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tools/distpartitioning/convert_partition.py [hunk 1]
```diff
@@ -183,7 +185,7 @@ def create_dgl_object(graph_name, num_parts, \
 
     return compact_g2, node_map_val, edge_map_val, ntypes_map, etypes_map
 
-def create_metadata_json(graph_name, num_nodes, num_edges, num_parts, node_map_val, \
+def create_metadata_json(graph_name, num_nodes, num_edges, part_id, num_parts, node_map_val, \
```
Comment (zheng-da) [inline tools/distpartitioning/convert_partition.py:188]: i don't understand. you changed the interface of `create_metadata_json` but doesn't change the place where it is called. how does it work?
also, why is this change related to the bug.

File: tools/distpartitioning/globalids.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data):
     #form the outgoing message
     send_nodes = []
     for i in range(world_size):
-        send_nodes.append(torch.Tensor(global_nids_ranks[i]).type(dtype=torch.int64))
+        send_nodes.append(torch.from_numpy(global_nids_ranks[i]).type(dtype=torch.int64))
```
Comment (zheng-da) [inline tools/distpartitioning/globalids.py:53]: is it related to the bug?

Comment (kylasa) [reply tools/distpartitioning/globalids.py:53]: Yes. This is the bug on which I apparently spent a week debugging with the mag_lsc dataset. Originally the tensor was created, with global_nids, using "torch.Tensor()" function call. Please note that the argument here global_nids_ranks[i] is a numpy array. This works for smaller datasets where this list is on the smaller side and I have not noticed any mem. corruption issues. But when testing with mag_lsc dataset, right after this function call when we compare the actual tensor created vs. the data passed to this function call ... they are not identical. There was some wierd kind of mem. corruption hapenning when the input numpy array is very large and torch.Tensor() was used in creating the Tensor. 

This issue seems to be resolved, when torch.from_numpy() function call was used to create the Tensor. In this case for larger messages the data in the newly created tensor and the actual data passed to this function call are always correct.

File: tools/distpartitioning/convert_partition.py [hunk 3]
```diff
@@ -226,12 +228,12 @@ def create_metadata_json(graph_name, num_nodes, num_edges, num_parts, node_map_v
                      'ntypes': ntypes_map,
                      'etypes': etypes_map}
 
-    for part_id in range(num_parts):
+    for part_id in range(num_parts):
```
Comment (zheng-da) [inline tools/distpartitioning/convert_partition.py:231]: don't add space in the end.

Comment (kylasa) [reply tools/distpartitioning/convert_partition.py:231]: removed the space as suggested.

**Pull request merged**